File: java/src/test/java/VcypherTest.java
import com.vb.vcypher.Vcypher;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class VcypherTest {

  @Test
  public void testOutput() {
    Assertions.assertEquals(Vcypher.encode("BFC"), "81156121701");
    Assertions.assertEquals(Vcypher.encode("*()"), "8123614147318");
  }

  @Test
  public void testEdgeCasesOutput() {
    Assertions.assertEquals(Vcypher.encode(""), "");
  }

  @Test
  public void testNonEnglishLettersOutputs() {
    Assertions.assertEquals(Vcypher.encode("ľ"), "623");
    Assertions.assertEquals(Vcypher.encode("š"), "415");
    Assertions.assertEquals(Vcypher.encode("č"), "445");
    Assertions.assertEquals(Vcypher.encode("ť"), "5134");
    Assertions.assertEquals(Vcypher.encode("ž"), "712");
    Assertions.assertEquals(Vcypher.encode("ý"), "71");
    Assertions.assertEquals(Vcypher.encode("á"), "44");
    Assertions.assertEquals(Vcypher.encode("í"), "612");
    Assertions.assertEquals(Vcypher.encode("é"), "513");
  }
}

File: python/vcypher_test.py
import unittest
import Vcypher


class TestVcypher(unittest.TestCase):
    def test_output(self):
        self.assertEqual(Vcypher.vcypher("BFC"), "81156121701")
        self.assertEqual(Vcypher.vcypher("*()"), "8123614147318")

    def test_edge_cases_output(self):
        self.assertEqual(Vcypher.vcypher(""), "")

    def test_non_english_letters_output(self):
        self.assertEqual(Vcypher.vcypher("ľ"), "623")
        self.assertEqual(Vcypher.vcypher("š"), "415")
        self.assertEqual(Vcypher.vcypher("č"), "445")
        self.assertEqual(Vcypher.vcypher("ť"), "5134")
        self.assertEqual(Vcypher.vcypher("ž"), "712")
        self.assertEqual(Vcypher.vcypher("ý"), "71")
        self.assertEqual(Vcypher.vcypher("á"), "44")
        self.assertEqual(Vcypher.vcypher("í"), "612")
        self.assertEqual(Vcypher.vcypher("é"), "513")


if __name__ == '__main__':
    unittest.main()
